pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+        :meth:`~pywbem.WBEMConnection.OpenReferenceInstances`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          InstanceName (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the source instance.
+            If this object does specifies a namespace, it must equal the
+            value of the namespace parameter.
+            Its `host` attribute will be ignored.
+
+        See :meth:`~pywbem.WBEMConnection.References` for description of
+        ResultClass, Role, , FilterQueryLanguage, FilterQuery,
+        OperationTimeout, ContinueOnError, and MaxObjectCount parameters
+
+        Returns:
+
+            See :meth:`~pywbem.WBEMConnection.References` for description
+            of the returns.
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """
+
+        self._test_pull_operations_disabled()
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout, ContinueOnError)
+
+        instances = self.References(namespace, InstanceName,
+                                    ResultClass=ResultClass,
+                                    Role=Role,
+                                    IncludeClassOrigin=IncludeClassOrigin,
+                                    PropertyList=PropertyList)
+
+        return self._open_response(namespace, instances,
+                                   'PullInstancesWithPath',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
+
+    ####################################################################
+    #
+    #   OpenAssociatorInstancePaths server request
+    #
+    ####################################################################
+
+    def OpenAssociatorInstancePaths(self, namespace, InstanceName,
+                                    AssocClass=None, ResultClass=None,
+                                    Role=None, ResultRole=None,
+                                    FilterQueryLanguage=None, FilterQuery=None,
+                                    OperationTimeout=None, ContinueOnError=None,
+                                    MaxObjectCount=None):
+        # pylint: disable=invalid-name
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.OpenAssociatorInstancePaths`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE